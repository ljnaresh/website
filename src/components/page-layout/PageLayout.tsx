import * as React from 'react';
import { Container } from 'reactstrap';
import styled from 'styled-components';

const StyledContainerDiv = styled.div`
  @media (min-width: 1200px) {
    .container {
      max-width: 840px !important;
    }
  }
`;

const StyledFooter = styled.footer`
  height: 3rem;
  border-top: solid 0.1rem;
  border-top-color: rgba(0, 0, 0, 0.125);
`;

type PageLayoutProps = {
  children: React.ReactNode;
  className?: string;
};

const PageLayout = ({ children, className = '' }: PageLayoutProps) => {
  const currentYear = new Date().getFullYear();
  const classes = `d-flex flex-column min-vh-100 ${className}`.trim();

  return (
    <StyledContainerDiv>
      <Container className={classes}>
        <main className='flex-fill'>{children}</main>
        <StyledFooter className='footer text-center pt-2'>
          <span className='text-muted'>Â© {currentYear}</span>
        </StyledFooter>
      </Container>
    </StyledContainerDiv>
  );
};

export default PageLayout;
