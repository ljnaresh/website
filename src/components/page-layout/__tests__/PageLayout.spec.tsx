import React from 'react';
import renderer from 'react-test-renderer';

import PageLayout from '../PageLayout';

describe('PageLayout', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PageLayout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
