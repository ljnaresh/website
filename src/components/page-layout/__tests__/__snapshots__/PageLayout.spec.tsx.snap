// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders correctly 1`] = `
<div
  className="d-flex flex-column min-vh-100 container"
>
  <main
    className="flex-fill"
  />
  <footer
    className="footer text-center pt-2"
  >
    <span
      className="text-muted"
    >
      Â© 
      2022
    </span>
  </footer>
</div>
`;
